src/controllers/time-slot.controller.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiSecurity, ApiTags } from '@nestjs/swagger';
+import { PatchingTimeSlotDTO } from 'src/dto/patching-time-slot.dto';
+import { TimeSlotDTO } from '../dto/time-slot.dto';
+import { GroupOrganizationAdminGuard } from '../guards/group-organization-admin.guard';
+import { JWTGuard } from '../guards/jwt.guard';
+import { OrganizationAdminGuard } from '../guards/organization-admin.guard';
+import { TimeSlotService } from '../services/time-slot.service';
+
+@Controller('timeslots')
+@ApiTags('TimeSlot')
+export class TimeSlotController {
+  constructor(private readonly timeSlotService: TimeSlotService) {}
+
+  @Post('/group/:groupId')
+  @UseGuards(JWTGuard, GroupOrganizationAdminGuard)
+  @ApiSecurity('Bearer')
+  createTimeSlot(@Param('groupId') groupId: string, @Body() body: TimeSlotDTO) {
+    return this.timeSlotService.insertTimeSlot(groupId, body);
+  }
+
+  @Get('/group/:groupId')
+  @UseGuards(JWTGuard, GroupOrganizationAdminGuard)
+  @ApiSecurity('Bearer')
+  readAllGroupTimeSlot(@Param('groupId') groupId: string) {
+    return this.timeSlotService.getAllGroupTimeSlots(groupId);
+  }
+
+  @Get('/group/:groupId/:timeSlotId')
+  @UseGuards(JWTGuard, GroupOrganizationAdminGuard)
+  @ApiSecurity('Bearer')
+  readOneGroupTimeSlot(
+    @Param('groupId') groupId: string,
+    @Param('timeSlotId') timeSlotId: string,
+  ) {
+    return this.timeSlotService.getOneGroupTimeSlotById(timeSlotId);
+  }
+
+  @Patch('/group/:groupId/:timeSlotId')
+  @UseGuards(JWTGuard, GroupOrganizationAdminGuard)
+  @ApiSecurity('Bearer')
+  updateOneGroupTimeSlot(
+    @Param('groupId') groupId: string,
+    @Param('timeSlotId') timeSlotId: string,
+    @Body() body: PatchingTimeSlotDTO,
+  ) {
+    return this.timeSlotService.updateOneGroupTimeSlotById(
+      { groupId, timeSlotId },
+      { ...body },
+    );
+  }
+
+  @Delete('/:timeSlotId')
+  @UseGuards(JWTGuard, OrganizationAdminGuard)
```

Comment:
OrganizationAdminGuard doesn't work without `:organizationId`

Reply:
You should create a guard for this and `time-slot.controller.ts:42`

Reply:
The DB request is made with a `groupId` not with a `organizationId `.

<img width="655" alt="Screenshot 2022-02-20 at 14 49 49" src="https://user-images.githubusercontent.com/32927775/154845788-e4b5109a-fb99-42dc-b645-9f21d53eda5e.png">

Reply:
But actually you don't have `groupId` 😆